features/dd-sdk-android-webview/src/main/kotlin/com/datadog/android/webview/internal/replay/WebViewReplayEventConsumer.kt

```diff
@@ -0,0 +1,123 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.webview.internal.replay
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.api.context.DatadogContext
+import com.datadog.android.api.feature.Feature
+import com.datadog.android.api.feature.FeatureScope
+import com.datadog.android.api.feature.FeatureSdkCore
+import com.datadog.android.core.InternalSdkCore
+import com.datadog.android.webview.internal.WebViewEventConsumer
+import com.datadog.android.webview.internal.rum.TimestampOffsetProvider
+import com.datadog.android.webview.internal.rum.WebViewRumEventContextProvider
+import com.datadog.android.webview.internal.rum.domain.RumContext
+import com.google.gson.JsonObject
+import java.lang.IllegalStateException
+import java.lang.NumberFormatException
+import java.lang.UnsupportedOperationException
+
+internal class WebViewReplayEventConsumer(
+    private val sdkCore: FeatureSdkCore,
+    webViewId: Long,
+    internal val offsetProvider: TimestampOffsetProvider,
+    private val contextProvider: WebViewRumEventContextProvider,
+    private val webViewReplayEventMapper: WebViewReplayEventMapper =
+        WebViewReplayEventMapper(webViewId)
+) : WebViewEventConsumer<JsonObject> {
+
+    override fun consume(event: JsonObject) {
+        val datadogContext = (sdkCore as InternalSdkCore).getDatadogContext()
+        val sessionReplayFeatureScope = sdkCore.getFeature(Feature.SESSION_REPLAY_FEATURE_NAME)
+        if (datadogContext != null && sessionReplayFeatureScope != null) {
+            sendEventThroughBus(datadogContext, event, sessionReplayFeatureScope)
+        }
+    }
+
+    private fun sendEventThroughBus(
+        datadogContext: DatadogContext,
+        event: JsonObject,
+        sessionReplayFeatureScope: FeatureScope
+    ) {
+        val rumContext = contextProvider.getRumContext(datadogContext)
```

Comment:
since we are sending the message through the message bus, wouldn't it be simpler than to do the mapping inside `SessionReplayFeature`? as a benefit there we can get `DatadogContext` directly from `withWriteContext`.

Reply:
nope because the interest is to add this context as fast as possible. By the time it reaches the `SessionReplayFeature` the context might already change. Also is following the same approach as for RUM events.